internal/controllers/common_controller.go

```diff
@@ -95,3 +120,46 @@ func setDeprecationStatusesUnknown(conditions *[]metav1.Condition, message strin
 		})
 	}
 }
+
+func updateStatusUnpackFailing(status *ocv1alpha1.ClusterExtensionStatus, err error) error {
+	status.InstalledBundle = nil
+	apimeta.SetStatusCondition(&status.Conditions, metav1.Condition{
+		Type:    rukpakv1alpha2.TypeUnpacked,
+		Status:  metav1.ConditionFalse,
+		Reason:  rukpakv1alpha2.ReasonUnpackFailed,
+		Message: err.Error(),
+	})
+	return err
+}
+
+// TODO: verify if we need to update the installBundle status or leave it as is.
+func updateStatusUnpackPending(status *ocv1alpha1.ClusterExtensionStatus, result *source.Result, generation int64) {
+	status.InstalledBundle = nil
+	apimeta.SetStatusCondition(&status.Conditions, metav1.Condition{
+		Type:               rukpakv1alpha2.TypeUnpacked,
```

Comment:
Same for the other two unpack status conditions below. 
```suggestion
		Type:               ocv1alpha1.TypeUnpacked,
```

Reply:
This is a status condition, didn't we say that another PR will deal with that stuff? (Although the above suggestion is correct.)

Reply:
#896